### ISSUE TITLE
1.10.1 issue with mixing TLS PASSTHROUGH and SIMPLE modes

### ISSUE THREAD

**Issue by maciekm** [istio/istio#33405]
**Bug description**
given a mix of gateways with TLS PASSTHROUGH and SIMPLE modes: 
```yaml
apiVersion: networking.istio.io/v1beta1
kind: Gateway
metadata:
  name: gateway-private
  namespace: istio-system
spec:
  selector:
    app: istio-ingressgateway
  servers:
  - hosts:
    - '*.example.com'
    port:
      name: http
      number: 80
      protocol: HTTP
    tls:
      httpsRedirect: true
  - hosts:
    - '*.example.com'
    port:
      name: https
      number: 443
      protocol: HTTPS
    tls:
      credentialName: wildcard-private-cert
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1beta1
kind: Gateway
  name: gateway-foo
  namespace: istio-system
spec:
  selector:
    app: istio-ingressgateway
  servers:
  - hosts:
    - foo.example.com
    port:
      name: http
      number: 80
      protocol: HTTP
    tls:
      httpsRedirect: true
  - hosts:
    - foo.example.com
    port:
      name: tls-foo
      number: 443
      protocol: TLS
    tls:
      mode: PASSTHROUGH
```
in insto **1.10.1** `istiod` complains:
```
2021-06-11T21:31:51.074820Z	warn	ads	ADS:LDS: ACK ERROR router~10.52.1.100~istio-ingressgateway-internal-6bc6b9cf87-q96jl.istio-system~istio-system.svc.cluster.local-171 Internal:Error adding/updating listener(s) 0.0.0.0_8443: duplicate listener 0.0.0.0_8443 found
```
and `istio-ingressgateway` complains:
```
2021-06-11T21:55:03.277665Z	warn	Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 2 successful, 0 rejected; lds updates: 0 successful, 2 rejected
```
proxy-config is:
```
$ istioctl proxy-config listeners istio-ingressgateway-xyz.istio-system
ADDRESS PORT  MATCH                  DESTINATION
0.0.0.0 8080  ALL                    Route: http.80
0.0.0.0 8443  SNI: foo.example.com   Cluster: outbound|443||foo-server.foo.svc.cluster.local
0.0.0.0 15021 ALL                    Inline Route: /healthz/ready*
0.0.0.0 15090 ALL                    Inline Route: /stats/prometheus*
```
removing  the `PASSTHROUGH` gateway makes the error go away.

in istio **1.9.3** it is working as expected and proxy-config is:
```
ADDRESS PORT  MATCH                                           DESTINATION
0.0.0.0 8080  ALL                                             Route: http.80
0.0.0.0 8443  SNI: *.example.com                              Route: https.443.https.gateway-private.istio-system
0.0.0.0 8443  SNI: foo.example.com                            Cluster: outbound|443||foo-server.foo.svc.cluster.local
0.0.0.0 15021 ALL                                             Inline Route: /healthz/ready*
0.0.0.0 15090 ALL                                             Inline Route: /stats/prometheus*
```

[x] Networking

**Expected behavior**

**1.10.1** `istio-ingressgateway` should be able to start with a mix of TLS PASSTHROUGH and SIMPLE mode gateways- as it does with istio **1.9.3**. It's also not an issue in istio **1.9.5**.

**Steps to reproduce the bug**
see -> Bug description

**Version** 
```
client version: 1.9.1
control plane version: 1.10.1
data plane version: 1.10.1 (6 proxies), 1.9.3 (15 proxies)
```

**How was Istio installed?**
istio operator

**Environment where the bug was observed (cloud vendor, OS, etc)**
eks

**Comment by howardjohn** [comment]
Do you have other Gateways and/or multiple services selecting the same gateway deployment?

**Comment by howardjohn** [comment]
@ramaraochavali this might be related to some of the changes we made in MergeGateway?

**Comment by ramaraochavali** [comment]
Quite possible. I will take a look.

**Comment by maciekm** [comment]
thanks @howardjohn and @ramaraochavali for looking into it. 

> Do you have other Gateways

Yes, a few, and they all work as expected. It's only that single TLS PASSTHROUGH gateway causing issues in istio 1.10.x. Removing that single  TLS PASSTHROUGH gateway makes it all work again.

> multiple services selecting the same gateway deployment

We have a single service (LoadBalancer) pointing to gateway deployment.

**Comment by klarose** [comment]
We're also running in to this. It is blocking us from upgrading to 1.10 from 1.9.3. 1.9 is no longer supported, so we'd like to upgrade as soon as possible.

When will this fix make it into a 1.10 release? It doesn't appear to be in 1.10.2, which was released after the fix was made.

**Comment by ramaraochavali** [comment]
@klarose Sorry. I could not get time to back port that to 1.10 as it needed manual merge. I will do it next couple of days

**Comment by ramaraochavali** [comment]
@klarose https://github.com/istio/istio/pull/34125 - Here is the manual backport.

**Comment by dskatz** [comment]
We are also running into this issue in 1.10.3.  When can we expect #33411 to be included in a release?

**Comment by howardjohn** [comment]
The fix will be in 1.10.4
